### ISSUE TITLE
Mobile Safari & Chrome on iOS 15.1 crash when publishing an H264 local video track

### ISSUE THREAD

**Issue by Bumbolio** [twilio/twilio-video.js#1611]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**
Here is a live example using twilio-video 2.18
https://switcher-video-chat.switcher-studio.workers.dev/3ca560d9-dda0-4f7e-bf34-8bf720fe34a8

**Expected behavior:**
When connecting to an H264 only Group Room, the local video track should be published successfully while using Safari or Chrome on iOS 15.1 (beta).

**Actual behavior:**
When connecting to an H264 only Group Room, if a local video track is provided when connecting while using Safari or Chrome on iOS 15.1 (beta), the browser tab crashes and the browser reloads the page. 

**Additional Information**
This is happening on the latest iOS 15.1 beta which appears to be the release candidate for iOS 15.1. There is a [Webkit bug filed](https://bugs.webkit.org/show_bug.cgi?id=232006) for a similar issue that has already been patched, but it's not clear if this will make it into the final release of iOS 15.1. According to the bug filed with Webkit, it appears to crash the tab when switching the video track off or on, but we're not doing that in our example. The tab crashes as soon as we connect to the room and the local video track is published. 

I was also able to reproduce this bug in the twilio-video-app-react reference project. 

**Software versions:**

 - [x] Browser(s): Safari & Chrome on iOS 15.1 (beta)
 - [x] Operating System: iOS 15.1 (beta)
 - [x] twilio-video.js: 2.18
 - [x] Third-party libraries (e.g., Angular, React, etc.): React

**Comment by aponski** [comment]
@twilio any update on this issue?

**Comment by aponski** [comment]
As a workaround I've set  `preferredVideoCodecs: ['VP8']`

**Comment by leslie-heinzen** [comment]
> @twilio any update on this issue?

Based on our own testing, this appears to be resolved in the iOS 15.2 beta.

**Comment by MEGApixel23** [comment]
> As a workaround I've set `preferredVideoCodecs: ['VP8']`

Thank you for the suggestion, but it doesn't work for us. It still crashes.

**Comment by taylor1791** [comment]
We used the `preferredVideoCodecs: ['VP8']` workaround successfully in two different applications.

```js
import TwilioVideo from 'twilio-video';

await TwilioVideo.connect(token, {
  name,
  preferredVideoCodecs: [ 'VP8' ]
});
```

**Comment by MEGApixel23** [comment]
> We used the `preferredVideoCodecs: ['VP8']` workaround successfully in two different applications.
> 
> ```js
> import TwilioVideo from 'twilio-video';
> 
> await TwilioVideo.connect(token, {
>   name,
>   preferredVideoCodecs: [ 'VP8' ]
> });
> ```

That's how we tried to apply suggested fix:
```js
import { connect, LocalVideoTrack, createLocalTracks } from 'twilio-video'

connect(data.token, {
  tracks,
  name: roomName,
  networkQuality: {
    local: 3
  },
  preferredVideoCodecs: [{ codec: 'VP8', simulcast: false }]
})
```

Actually, I've tried all of the available combinations from their docs: https://www.twilio.com/docs/video/tutorials/developing-high-quality-video-applications#collaboration-mode

Additionally I've tried several twilio-video package versions starting from
```"twilio-video": "^2.16.0",``` up to the latest.

Unfortunatelly it still crashed on my iPhone 12 iOS 15.1 😞

**Comment by MisterJimson** [comment]
Maybe we should update the title and pin the issue? Its not beta only anymore and this is a pretty big issue.

**Comment by charliesantos** [comment]
>Unfortunatelly it still crashed on my iPhone 12 iOS 15.1 😞

@MEGApixel23 are you still seeing the issue on iOS 15.2 beta?

**Comment by MEGApixel23** [comment]
> > Unfortunatelly it still crashed on my iPhone 12 iOS 15.1 😞
> 
> @MEGApixel23 are you still seeing the issue on iOS 15.2 beta?

We haven't tested with iOS 15.2 yet. Is there a safe way to do it considering that I have only one iPhone 😅 ?

**Comment by charliesantos** [comment]
> > > Unfortunatelly it still crashed on my iPhone 12 iOS 15.1 😞
> > 
> > 
> > @MEGApixel23 are you still seeing the issue on iOS 15.2 beta?
> 
> We haven't tested with iOS 15.2 yet. Is there a safe way to do it considering that I have only one iPhone 😅 ?

@MEGApixel23 Not sure if there's other more reliable way to test. Can you please try it on our [quick deploy react app](https://github.com/twilio/twilio-video-app-react)? This is configured to use VP8. If you are able to reproduce the crash with the quick deploy app using VP8, please file a separate ticket.

**Comment by charliesantos** [comment]
I verified and this seems to have been resolved on iOS 15.2 beta

**Comment by vatson** [comment]
@charliesantos is right. iOS 15.2 beta works well with h264

**Comment by Bikash-lama** [comment]
> We used the `preferredVideoCodecs: ['VP8']` workaround successfully in two different applications.
> 
> ```js
> import TwilioVideo from 'twilio-video';
> 
> await TwilioVideo.connect(token, {
>   name,
>   preferredVideoCodecs: [ 'VP8' ]
> });
> ```

This workaround not working for me. Any feedback would be appreciated.

**Comment by charliesantos** [comment]
@Bikash-lama can you please try on iOS 15.2 beta and see if you're still having this issue? If you're still seeing it, please submit a different ticket. It may be something that is not related to h264

**Comment by MEGApixel23** [comment]
@charliesantos Confirmed working on iOS 15.2 beta 3. However, the same app crashes on iOS 15.1 😞

**Comment by leslie-heinzen** [comment]
> @charliesantos Confirmed working on iOS 15.2 beta 3. However, the same app crashes on iOS 15.1 😞

iOS 15.2 will probably be generally available around the end of this month. Apple is not going to patch this in iOS 15.1.

**Comment by juneroy1** [comment]
Bro. 

> We used the `preferredVideoCodecs: ['VP8']` workaround successfully in two different applications.
> 
> ```js
> import TwilioVideo from 'twilio-video';
> 
> await TwilioVideo.connect(token, {
>   name,
>   preferredVideoCodecs: [ 'VP8' ]
> });
> ```

this works in me in IOS15.1 BRO
![image](https://user-images.githubusercontent.com/27282831/143774396-16baebb1-c041-47e0-b2dc-611f9b38cf59.png)

**Comment by Bumbolio** [comment]
Yes,if you can get away with using VP8 then this is a good workaround. Our application requires H264, so the only solution for us is to wait for iOS 15.2 to be generally available.

**Comment by Bikash-lama** [comment]
> Yes,if you can get away with using VP8 then this is a good workaround. Our application requires H264, so the only solution for us is to wait for iOS 15.2 to be generally available.

it's still not working for me. I am currently using Twilio 2.8.0 CDN approach rather than a package. May be thats why the workaround is not working for me.

CDN url: https://media.twiliocdn.com/sdk/js/video/releases/2.8.0/twilio-video.min.js

**Comment by juneroy1** [comment]
> > Yes,if you can get away with using VP8 then this is a good workaround. Our application requires H264, so the only solution for us is to wait for iOS 15.2 to be generally available.
> 
> it's still not working for me. I am currently using Twilio 2.8.0 CDN approach rather than a package. May be thats why the workaround is not working for me.
> 
> CDN url: https://media.twiliocdn.com/sdk/js/video/releases/2.8.0/twilio-video.min.js

I think you have to update your SDK bro

**Comment by charliesantos** [comment]
Hi @darrenmhill , unfortunately we cannot really do anything on the sdk. Your application needs to use vp8 for now if possible.

**Comment by arun3528** [comment]
HI Team we are seeing another issue where On Android 12 H264 preferred codec does not encode video properly 

tried the same url on Pixcel 3a 

https://switcher-video-chat.switcher-studio.workers.dev/3ca560d9-dda0-4f7e-bf34-8bf720fe34a8

![WhatsApp Image 2021-12-07 at 7 03 52 PM](https://user-images.githubusercontent.com/3895486/145038679-cdb7f75f-2f2f-457a-94aa-1a4d654f8fbc.jpeg)
Result 

Does not happen on android 11.0

**Comment by PikaJoyce** [comment]
Hi @arun3528 

Thank you for your comment. We currently have an [open issue](https://github.com/twilio/twilio-video.js/issues/1627) with regards to your comment! To forward that information from that thread over, we don't have any workarounds currently for this issue. So we're currently waiting on a Chrome fix which should be coming soon.

Thanks,
Joyce

**Comment by charliesantos** [comment]
We verified this is fixed on iOS 15.2.
